woodwork/logical_types.py

```diff
@@ -71,7 +72,16 @@ def transform(self, series):
         return series
 
     def validate(self, series, return_invalid_values=True):
-        return
+        """Validates that a logical type is consistent with the series dtype. Performs additional type
+        specific validation, as required."""
+        valid_dtype = self._get_valid_dtype(type(series))
+        if valid_dtype != str(series.dtype):
+            raise ValueError(
+                f"Cannot initialize Woodwork. Series dtype '{series.dtype}' is "
+                f"incompatible with {self.type_string} dtype. Try converting series "
+                f"dtype to '{valid_dtype}' before initializing or use the "
+                "woodwork.init_series function to initialize."
```

Comment:
changed the error to TypeValidationError

Reply:
Let's also change the message here at the `LogicalType.validate` method to say 

```
                f"Series dtype '{series.dtype}' is incompatible with {self.type_string} dtype."
```
So that it can be used outside of Woodwork initialization